### ISSUE TITLE
[Extension Proposal]  Quarkus Gradle UI

### ISSUE THREAD

**Issue by rmanibus** [quarkusio/quarkus#35741]
### Description

Interested in this extension, please +1 via the emoji/reaction feature of GitHub (top right).

I am opening this to initiate the discussion around piloting build tools from the web UI.

I have a POC working with gradle that allow to list all project tasks and execute them:

https://github.com/rmanibus/quarkus-gradle-ui

Here are a couple of features I can see fit for this extension: 

- executing tasks from the UI
- view dependencies versions
- upgrade dependencies (update interactive style)

Some open questions:

- Currently it is hard to get info about the current build (no clear way to get the project directory for instance), what would it take to expose something like core/deployment/src/main/java/io/quarkus/deployment/dev/DevModeContext.java ?
- Could this be built in a tool agnostic way (working for both maven and gradle) ?

### Repository name

quarkus-gradle-ui

### Short description

Gradle Dashboard in quarkus Dev UI

### Repository Homepage URL

https://quarkiverse.github.io/quarkiverse-docs/quarkus-gradle-ui/dev/

### Repository Topics

- quarkus-extension
- 
...

### Team Members

-

**Comment by gastaldi** [comment]
Interesting, how hard would it be to support Maven as well?

**Comment by aloubyansky** [comment]
FYI @snazy

**Comment by maxandersen** [comment]
jbang?  :)

(semi) kidding aside - not fan of making it gradle (or maven or jbang) specific if it is in core; but having a devui into the build info available no matter what build system with option to do certain build specific task that wont break devmode that would get a +1 from me.

**Comment by rmanibus** [comment]
So are you saying this should live in core rather than quarkiverse if we manage to make it framework agnostic ?

**Comment by gastaldi** [comment]
I don't think so, it can still live as a Quarkiverse extension, which would benefit from faster release cycles

**Comment by rmanibus** [comment]
does anyone know if it is possible to access the DevModeContext (passed by the plugin) from a Buildstep ?

**Comment by rmanibus** [comment]
Also, what is the best way to detect the build tool (beside checking if there is a pom in the project) ? 
Could we imagine using Capabilities for this ?
Or maybe we could introduce a new buildItem carrying metadata about the build environment ?

**Comment by aloubyansky** [comment]
> does anyone know if it is possible to access the DevModeContext (passed by the plugin) from a Buildstep ?

No, it's not exposed at the moment. What specifically are you looking for?

> Or maybe we could introduce a new buildItem carrying metadata about the build environment ?

There is `BuildSystemTargetBuildItem`, which doesn't have this info but possibly could be enhanced.
There is also `CurateOutcomeBuildItem` that exposes `ApplicationModel` and then there is `io.quarkus.devtools.project.BuildTool` that exposes `BuildTool fromProject(Path projectDirPath)`. The project dir could be taken from `ApplicationModel.getAppArtifact().getWorkspaceModule().getModuleDir()`, which will work in dev mode.

**Comment by rmanibus** [comment]
thanks a lot, I think this is what I need !
